src/types-and-traits.rst

```diff
@@ -1995,6 +1995,41 @@ obtained as follows:
    #. :dp:`fls_ju4ypa5ysga0`
       Continue with the next source :t:`type`.
 
+
+.. _fls_emcNJzl2tHSA:
+
+Structural Equality
+~~~~~~~~~~~~~~~~~~~
+
+.. rubric:: Legality Rules
+
+:dp:`fls_uVTpA7gbLCYX`
+A :t:`type` is a :t:`structurally equal` when its :t:`values` can be compared
+for equality by structure.
+
+:dp:`fls_2DZAP6JJjJ9h`
+The following :t:`[type]s` are structurally equal:
+
+* :dp:`fls_emcNJzl2tHSA`
+  :c:`Bool`, :c:`char`, :t:`[function pointer type]s`, :t:`[integer type]s`,
+  :c:`str`,  and :t:`[raw pointer type]s`.
+
+* :dp:`fls_HpWSAfaTA1Dz`
+  An :t:`abstract data type` if it implements the :t:`[trait]s`
+  :std:`core::cmp::Eq` and :std:`core::cmp::PartialEq` via the use of the
+  :t:`[derive macro]s` :std:`core::cmp::Eq` and :std:`core::cmp::PartialEq`.
```

Comment:
Oh, it is no longer needed to use attribute derive?

Reply:
Well, you need to use it to invoke the `derive macros`. But my previous phrasing was incorrect as that would imply you could write a custom `derive macro` that implements them which is not the case, only the tool builtin derives have the desired effect

Reply:
Gotcha.